### PR TITLE
Added Subset Sum Problem

### LINKED ISSUE #889: Adding Subset Sum problem using DP in C++, Java and Python

### PR DESCRIPTION
**Pull request by singh-shreya6**
Fixes #889 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:
-  Added Subset Sum Problem using Dynamic rogramming

#### Languages Used:
- C++
-  Java
- Python

#### Files Added:
- Subset_Sum.cpp
- Subset_Sum.java
- Subset_Sum.py

### CODE REVIEW

File: Subset_Sum/Subset_Sum.cpp [hunk 1]
```diff
@@ -0,0 +1,74 @@
+/* Subset Sum Problem: Given an array of elements and a sum, is it possible to form the given sum using some or all of the elements of array i.e using any subset of given array.
```
Comment (SakshiShreya) [inline Subset_Sum/Subset_Sum.cpp]: Divide the comment into separate lines instead of using a single line.

File: Subset_Sum/Subset_Sum.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    /* The last entry in table gives the answer by evaluating all the elements of array to 
+    compute the required sum */
+    return dp[size][sum];
+}
+
+// Driver function
+int main()
+{
+    ll arr[] = {2, 3, 5, 1, 6, 8, 7};
+    ll size = sizeof(arr)/sizeof(arr[0]);
```
Comment (SakshiShreya) [inline Subset_Sum/Subset_Sum.cpp]: Add space around operators.

File: Subset_Sum/Subset_Sum.cpp [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    ll sum = 12;
+
+    // Call to SubsetSum function
+    bool ans = SubsetSum(arr, size, sum);
+
+    // if ans evaluates to false, no such subset present
+    // if ans evaluates to true , subset with given sum present
+    if (!ans)
+        cout<<"No subset adds upto the given sum"<<endl;
```
Comment (SakshiShreya) [inline Subset_Sum/Subset_Sum.cpp]: Add space around `<<`

File: Subset_Sum/Subset_Sum.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    for i in range(0, size+1):
+        for j in range(0, sum+1):
+
+            # We can always form the sum 0, by taking 0 instance of the element
+            if j == 0 :
+                dp[i][0] = True
+            
+            if i > 0 :
+                ''' when the value of current element is less than the sum j, we have two 
+                options, either take it and include the element in answer or leave it and take
```
Comment (SakshiShreya) [inline Subset_Sum/Subset_Sum.py]: Divide the comment into separate lines. Try not to exceed 80 characters per line (including the indentation).

Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.py]: Made all changes. Please review.

File: Subset_Sum/Subset_Sum.cpp [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+Example: 1, 7, 3, 5, 6 and sum = 14, it is possible to
+obtain the sum 14 by taking the elements 1, 7 and 6 */
+
+#include <bits/stdc++.h>
+using namespace std;
+typedef long long int ll;
+
+// Function to compute whether a subset of array exists
+// having the given sum
+bool SubsetSum(ll arr[], ll size, ll sum)
```
Comment (sridharjajoo) [inline Subset_Sum/Subset_Sum.cpp]: function names in c++ are always camel case. The above style of naming is done in C#

Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.cpp]: done

File: Subset_Sum/Subset_Sum.py [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+Example: 1, 7, 3, 5, 6 and sum = 14, it is possible to obtain 
+the sum 14 by taking the elements 1, 7 and 6 '''
+
+# Function to compute whether a subset of array exists 
+# having the given sum
+def subsetSum(arr, size, sum) :
+
+    # A 2D Matrix containing boolean values, true and false
+    dp = []
+    for i in range(0, size+1):
```
Comment (SakshiShreya) [inline Subset_Sum/Subset_Sum.py]: Add spaces around all operators in this file. Other files are fine.

Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.py]: Done

File: Subset_Sum/Subset_Sum.cpp [hunk 7]
```diff
@@ -0,0 +1,83 @@
+/* Subset Sum Problem: Given an array of elements and a sum,
```
Comment (SakshiShreya) [inline Subset_Sum/Subset_Sum.cpp]: Use 4 space indentation.

File: Subset_Sum/Subset_Sum.java [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+                        we cannot include this element and hence take the
+			previous best*/
+                        dp[i][j] = dp[i - 1][j];
+                }
+            }
+        }
+
+        /* The last entry in table gives the answer by
+        evaluating all the elements of array to compute
+	the required sum */
```
Comment (SakshiShreya) [inline Subset_Sum/Subset_Sum.java]: This line is indented with tab. Use 8 spaces for this line.

File: Subset_Sum/Subset_Sum.cpp (hunk 7 above)
Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.cpp]: @SakshiShreya I don't see the need of 4 space indentation at the start of comments.  Indentation is provided for clarity at the start of a block.

File: Subset_Sum/Subset_Sum.java (hunk 8 above)
Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.java]: It is exactly 8 spaces.

Comment (SakshiShreya) [reply Subset_Sum/Subset_Sum.java]: Line 55 is a tab

File: Subset_Sum/Subset_Sum.cpp (hunk 7 above)
Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.cpp]: Please clarify the need of 4 space indentation here @SakshiShreya .

File: Subset_Sum/Subset_Sum.java (hunk 8 above)
Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.java]: It is exactly in line with the above statements.

File: Subset_Sum/Subset_Sum.cpp (hunk 7 above)
Comment (jainaman224) [reply Subset_Sum/Subset_Sum.cpp]: Actually she meant throughout the file.

File: Subset_Sum/Subset_Sum.java (hunk 8 above)
Comment (jainaman224) [reply Subset_Sum/Subset_Sum.java]: It is recommended to use space or tab not both. We prefer to use spaces instead of tabs as well.

Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.java]: I have made the recommended changes. Please review.

File: Subset_Sum/Subset_Sum.cpp (hunk 7 above)
Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.cpp]: Done.

File: Subset_Sum/Subset_Sum.java (hunk 8 above)
Comment (singh-shreya6) [reply Subset_Sum/Subset_Sum.java]: @SakshiShreya Please review.

**Pull request merged**